wger/weight/helpers.py

```diff
@@ -43,21 +43,36 @@ def parse_weight_csv(request, cleaned_data):
     # csv.reader expects a file-like object, so use StringIO
     parsed_csv = csv.reader(six.StringIO(cleaned_data['csv_input']),
                             dialect)
-
+    distinct_weight_entries = []
     weight_list = []
     error_list = []
+
+    # Process the CSV items first
     for row in parsed_csv:
         try:
             parsed_date = datetime.datetime.strptime(row[0], cleaned_data['date_format'])
             parsed_weight = decimal.Decimal(row[1].replace(',', '.'))
-
-            weight_list.append(WeightEntry(date=parsed_date,
-                                           weight=parsed_weight,
-                                           user=request.user))
+            duplicate_date_in_db = WeightEntry.objects.filter(date=parsed_date,
+                                                              user=request.user)
```

Comment:
Since there can only be one WeightEntry object per user per date, it would be better to write ``.filter(...)exists()``, that returns a boolean and is more efficient DB-wise